src/main/java/bio/terra/workspace/service/resource/controlled/ControlledResourceService.java

```diff
@@ -92,34 +92,24 @@ public ControlledResource validateControlledResourceAndAction(
     return controlledResource;
   }
 
-  public ControlledResource syncCreateControlledResource(
-      ControlledResource resource,
+  /** Starts a create controlled bucket resource, blocking until its job is finished. */
+  public ControlledGcsBucketResource syncCreateBucket(
+      ControlledGcsBucketResource resource,
       ApiGcpGcsBucketCreationParameters creationParameters,
       List<ControlledResourceIamRole> privateResourceIamRoles,
       AuthenticatedUserRequest userRequest) {
-    ApiJobControl syncJobControl = new ApiJobControl().id(UUID.randomUUID().toString());
-    String jobId =
-        createControlledResource(
-            resource,
-            creationParameters,
-            privateResourceIamRoles,
-            syncJobControl,
-            null,
-            userRequest);
-    jobService.waitForJob(jobId);
-    JobResultOrException<ControlledResource> jobResult =
-        jobService.retrieveJobResult(jobId, ControlledResource.class, userRequest);
-    if (jobResult.getException() != null) {
-      throw jobResult.getException();
-    }
-    return jobResult.getResult();
+    JobBuilder jobBuilder =
+        commonCreationJobBuilder(
+                resource, privateResourceIamRoles, UUID.randomUUID().toString(), null, userRequest)
+            .addParameter(ControlledResourceKeys.CREATION_PARAMETERS, creationParameters);
+    return jobBuilder.submitAndWait(ControlledGcsBucketResource.class);
   }
 
-  public String createControlledResource(
+  /** Create a JobBuilder for creating controlled resources with the common parameters populated. */
+  private JobBuilder commonCreationJobBuilder(
       ControlledResource resource,
-      ApiGcpGcsBucketCreationParameters creationParameters,
       List<ControlledResourceIamRole> privateResourceIamRoles,
-      ApiJobControl jobControl,
+      String jobId,
```

Comment:
This is fine for now, but we will want the full JobControl object for the JobService when we implement pub/sub completions.

Reply:
done.